include/alpaka/core/Assert.hpp

```diff
@@ -15,7 +15,13 @@
 #include <type_traits>
 
 
-#define ALPAKA_ASSERT(EXPRESSION) assert(EXPRESSION)
+#if defined(__SYCL_DEVICE_ONLY__)
```

Comment:
Maybe this check should go inside `ALPAKA_ASSERT_OFFLOAD` ?

Reply:
Indeed, thanks for catching this.